pkg/capabilities/consensus/ocr3/datafeeds/feeds_aggregator.go

```diff
@@ -47,25 +48,31 @@ var _ types.Aggregator = (*dataFeedsAggregator)(nil)
 // Metadata is a map of feedID -> (timestamp, price) representing onchain state (see DataFeedsOutcomeMetadata proto)
 func (a *dataFeedsAggregator) Aggregate(previousOutcome *types.AggregationOutcome, observations map[ocrcommon.OracleID][]values.Value) (*types.AggregationOutcome, error) {
 	// find latest valid Mercury report for each feed ID
-	latestReportPerFeed := make(map[mercury.FeedID]mercury.Report)
+	latestReportPerFeed := make(map[mercury.FeedID]mercury.FeedReport)
 	for nodeID, nodeObservations := range observations {
 		// we only expect a single observation per node - new Mercury data
-		if len(nodeObservations) == 0 {
+		if len(nodeObservations) == 0 || nodeObservations[0] == nil {
```

Comment:
@bolekk Hmmm... curious as to why this or was needed? Do we ever get nil values here?

Reply:
The answer to the above should be a code comment!

Reply:
In addition, what are the circumstances when a node would contribute more than one observation? Do we look at those warnings? What's the expected course of action when this happens?

Reply:
I somehow got it during tests. Does engine guarantee non-nil? A malicious node could send it. Doesn't hurt to have an extra check.

Reply:
`nil` would mean that a node didn't get an observation, right?

Reply:
It's different. If a node didn't send on observation during OCR protocol, it wouldn't appear in this map `observations map[ocrcommon.OracleID][]values.Value`.

I think a node could still send us a list of nils and the Outcome() implementation would pass it to the aggregator. I'm not sure if it makes sense to filter it out earlier (maybe a nil could make sense for other aggregators?)